src/datepicker-input/datepicker-input.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, Input } from "@angular/core";
+
+let nextId = 0;
+
+@Component({
+	selector: "ibm-date-picker-input",
+	template: `
+	<div class="bx--form-item">
+		<div class="bx--date-picker"
+		[ngClass]= "'bx--date-picker--'+type">
+			<div class="bx--date-picker-container">
+				<label [for]="id" class="bx--label">
+					{{label}}
+				</label>
+				<svg *ngIf="type == 'single'"
```

Comment:
Could this just check against `hasIcon`?

Reply:
no :(